gen.go

```diff
@@ -1269,7 +1270,8 @@ func (t *{{ .Name}}) UnmarshalCBOR(r io.Reader) error {
 
 	return doTemplate(w, gti, `
 		default:
-			return fmt.Errorf("unknown struct field %d: '%s'", i, name)
+			// Field doesn't exist on this type, so ignore it
+			cbg.ScanForLinks(r, func(cid.Cid){})
```

Comment:
We may want to make this opt-in (or not; I can't really think of anywhere where making it the default would be a problem)

Reply:
Also the method name is moderately odd for what we use it here, but I guess that's ok :shrug: